src/Parser/Properties/Property.ts

```diff
@@ -2,10 +2,12 @@ export default abstract class Property<T = string, P extends {[key: string]: str
     public abstract readonly key: string;
     public readonly value: T;
     public readonly parameters: P;
+    public readonly isNonStandard: boolean;
 
-    constructor (value: T, parameters: P) {
+    constructor (value: T, parameters: P, isNonStandard: boolean = false) {
```

Comment:
There is a short-hand notation for constructors when the argument name matches the target field name.

https://www.typescriptlang.org/docs/handbook/2/classes.html#constructors

This constructor would be written as:

```ts
constructor (public readonly value: T, public readonly parameters: P, public readonly isNonStandard: boolean = false) {}
```

Some linters complain about the empty constructor body which you can solve with an `/* empty on purpose */` comment or similar.

Also note that in this case if there is additional logic besides the fields you can still have regular code in the constructor, the body executes after field assignments.

Reply:
@passionate-bram Nice, I didn't know about constructor parameter promotion in TypeScript yet. I'll update this soon. Thanks for your help!